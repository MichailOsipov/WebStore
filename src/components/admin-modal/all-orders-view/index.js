export {AllOrdersView} from './all-orders-view';
