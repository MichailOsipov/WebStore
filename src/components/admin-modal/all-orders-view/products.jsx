import * as React from 'react';
import {ProductCard} from 'components/product-card';

export const Products = ({products}) => (
    <div>
        {products.map(({id, name, imageSrc}) => (
            <ProductCard
                key={id}
                name={name}
                imageSrc={imageSrc}
            />
        ))}
    </div>
);
