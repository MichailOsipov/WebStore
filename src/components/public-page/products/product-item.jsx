import * as React from 'react';
import {Button} from '@material-ui/core';

export const ProductItem = ({id, name, onOpenProductInformation}) => (
    <div key={id}>
        {name}
        <Button variant="contained" onClick={onOpenProductInformation}>Подробнее</Button>
    </div>
);
