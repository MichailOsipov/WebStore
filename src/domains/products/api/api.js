import {MOCK_PRODUCTS} from './mock-products';
import {MOCK_PRODUCT} from './mock-product';
import {MOCK_PRODUCTS_ADDED_TO_CART} from './mock-products-added-to-cart';
import {MOCK_ORDERS} from './mock-orders';

export const ProductsApi = {
    loadProducts: () => Promise.resolve(MOCK_PRODUCTS),
    loadProduct: productId => Promise.resolve(MOCK_PRODUCT),
    addProductToCart: productId => Promise.resolve(),
    loadProductsAddedToCart: () => Promise.resolve(MOCK_PRODUCTS_ADDED_TO_CART),
    removeProductFromCart: productId => Promise.resolve(),
    makeAnOrder: () => Promise.resolve(),
    loadOrders: () => Promise.resolve(MOCK_ORDERS)
};
