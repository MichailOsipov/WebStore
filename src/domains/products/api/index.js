export {ProductsApi} from './api';
export {MOCK_ALL_ORDERS} from './mock-all-orders';
export {MOCK_ORDERS} from './mock-orders';
export {MOCK_PRODUCT} from './mock-product';
export {MOCK_PRODUCTS_ADDED_TO_CART} from './mock-products-added-to-cart';
export {MOCK_PRODUCTS} from './mock-products';
