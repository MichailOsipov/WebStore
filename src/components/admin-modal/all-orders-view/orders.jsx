import * as React from 'react';
import {Button, Grid} from '@material-ui/core';
import {Products} from './products';

export const Orders = ({orders, confirmOrder}) => (
    <Grid container direction="column" spacing={40}>
        {orders.map(({id, products}) => (
            <Grid item key={id}>
                <Grid container direction="column" spacing={16}>
                    <Grid item>
                        <Products products={products} />
                    </Grid>
                    <Grid item>
                        <Button onClick={() => confirmOrder(id)} variant="contained">Подтвердить</Button>
                    </Grid>
                </Grid>
            </Grid>
        ))}
    </Grid>
);
