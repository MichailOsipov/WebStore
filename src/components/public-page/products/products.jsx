import * as React from 'react';
import {connect} from 'react-redux';
import {Grid, Typography} from '@material-ui/core';
import {getProducts, fetchProducts} from 'domains/products/state';
import {ProductItem} from './product-item';

export const Products = connect(
    state => ({
        products: getProducts(state)
    }),
    {
        fetchProducts
    }
)(
    class extends React.Component {
        componentDidMount() {
            this.props.fetchProducts();
        }

        render() {
            const {onOpenProductInformation, products} = this.props;
            return (
                <Grid container justify="center">
                    <Grid item>
                        <Typography variant="h6">Товары в наличии:</Typography>
                    </Grid>
                    <Grid container justify="center">
                        {products.map(({id, name, imageSrc}) => (
                            <Grid item xs={2} key={id}>
                                <ProductItem
                                    name={name}
                                    imageSrc={imageSrc}
                                    onOpenProductInformation={() => onOpenProductInformation(id)}
                                />
                            </Grid>
                        ))}
                    </Grid>
                </Grid>
            );
        }
    });
