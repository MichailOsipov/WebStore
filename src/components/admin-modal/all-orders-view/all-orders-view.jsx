import * as React from 'react';
import {Grid, Typography} from '@material-ui/core';
import {Orders} from './orders';

export const AllOrdersView = ({allOrders, confirmOrder}) => (
    <Grid container direction="column" spacing={16}>
        {allOrders.map(({user, orders}) => (
            <Grid item key={user}>
                <Grid container direction="column" spacing={16}>
                    <Grid item>
                        <Typography variant="h6">{user}</Typography>
                    </Grid>
                    <Grid item>
                        <Orders orders={orders} confirmOrder={confirmOrder} />
                    </Grid>
                </Grid>
            </Grid>
        ))}
    </Grid>
);
