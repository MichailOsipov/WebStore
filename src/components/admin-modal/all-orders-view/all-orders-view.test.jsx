import * as React from 'react';
import {shallow} from 'enzyme';
import {MOCK_ALL_ORDERS} from 'domains/products/api';
import {AllOrdersView} from './all-orders-view';

describe('AllOrdersView', () => {
    test('rendering', () => {
        const wrapper = shallow(
            <AllOrdersView allOrders={MOCK_ALL_ORDERS} confirmOrder={jest.fn()} />
        );
        expect(wrapper).toMatchSnapshot();
    });
});
