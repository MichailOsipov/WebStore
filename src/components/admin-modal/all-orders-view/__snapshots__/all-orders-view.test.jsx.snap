// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllOrdersView rendering 1`] = `
<WithStyles(Grid)
  container={true}
  direction="column"
  spacing={16}
>
  <WithStyles(Grid)
    item={true}
    key="Андрей"
  >
    <WithStyles(Grid)
      container={true}
      direction="column"
      spacing={16}
    >
      <WithStyles(Grid)
        item={true}
      >
        <WithStyles(Typography)
          variant="h6"
        >
          Андрей
        </WithStyles(Typography)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
      >
        <Orders
          confirmOrder={[MockFunction]}
          orders={
            Array [
              Object {
                "id": 3,
                "products": Array [
                  Object {
                    "id": 4,
                    "imageSrc": "https://media.tvzvezda.ru/news/vstrane_i_mire/content/201811191113-bhlk.htm/1.jpg",
                    "name": "Автомобиль",
                  },
                  Object {
                    "id": 5,
                    "imageSrc": "https://50.img.avito.st/640x480/3764525250.jpg",
                    "name": "Холодильник",
                  },
                ],
              },
              Object {
                "id": 4,
                "products": Array [
                  Object {
                    "id": 6,
                    "imageSrc": "https://zs-rus.ru/wp-content/uploads/2017/08/fcb45dbfbd1acf04a8965f251d76d573-1.jpg",
                    "name": "Лопата",
                  },
                ],
              },
            ]
          }
        />
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
    key="Никита"
  >
    <WithStyles(Grid)
      container={true}
      direction="column"
      spacing={16}
    >
      <WithStyles(Grid)
        item={true}
      >
        <WithStyles(Typography)
          variant="h6"
        >
          Никита
        </WithStyles(Typography)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
      >
        <Orders
          confirmOrder={[MockFunction]}
          orders={
            Array [
              Object {
                "id": 5,
                "products": Array [
                  Object {
                    "id": 7,
                    "imageSrc": "https://www.nv-lab.ru/images/upload/editor5036.jpg",
                    "name": "Стол",
                  },
                ],
              },
            ]
          }
        />
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
